tools/inventory-downloader/src/main.go

```diff
@@ -0,0 +1,211 @@
+	}
+
+	return response.Body, nil
+}
+
+func uploadToS3(cfg aws.Config, ctx context.Context, reader io.Reader, bucketName, objectKey string) error {
+	client := s3.NewFromConfig(cfg)
+	uploader := manager.NewUploader(client)
+
+	_, err := uploader.Upload(ctx, &s3.PutObjectInput{
+		Bucket: &bucketName,
+		Key:    &objectKey,
+		Body:   reader,
+	})
+	return err
+}
+
+func processFile(cfg aws.Config, ctx context.Context, objectKey string, fileUrl string) error {
+	reader, err := downloadFile(fileUrl)
+	if err != nil {
+		return fmt.Errorf("Error downloading file: %w", err)
+
+	}
+	defer reader.Close()
+
+	err = uploadToS3(cfg, ctx, reader, *bucketName, objectKey)
+	if err != nil {
+		return fmt.Errorf("Error uploading file to S3: %w", err)
+	}
+	return nil
+}
+
+var bucketName = flag.String("bucket", "joom-analytics-landing", "Target bucket to save files")
+var targetPrefix = flag.String("prefix", "storage-advisor/inventory", "Prefix to save files")
+var brokers = flag.String("brokers", "localhost:29092", "kafka brokers")
+var groupId = flag.String("group", "inventory-downloader-gburg-1", "kafka listener group")
+var topic = flag.String("topic", "", "kafka listener group")
+
+type Data struct {
+	ProjectId int
+	Payload   InventoryManifest
+}
+
+func processPayload(cfg aws.Config, ctx context.Context, inputManifest InventoryManifest, projectId int) error {
+	var err error
+	outputManifest := InventoryManifest{PartitionDate: inputManifest.PartitionDate}
+
+	for i, file := range inputManifest.Files {
+		filename, err := extractFilenameFromURL(file.Url)
+		if err != nil {
+			return fmt.Errorf("Error getting filename form URL: %w", err)
+		}
+
+		objectKey := fmt.Sprintf("%s/%s/%s/%s/%s/%s",
+			*targetPrefix,
+			strconv.Itoa(projectId),
+			inputManifest.Bucket,
+			inputManifest.PartitionDate,
+			"data",
+			filename)
+		log.Printf("(%d/%d) Downloading file to s3://%s/%s\n", i, len(inputManifest.Files), *bucketName, objectKey)
+		err = processFile(cfg, ctx, objectKey, file.Url)
+		if err != nil {
+			return fmt.Errorf("Error processing file: %w", err)
+		}
+		outputManifest.Files = append(outputManifest.Files, InventoryFile{Key: objectKey, MD5checksum: file.MD5checksum})
+	}
+
+	manifestKey := fmt.Sprintf("%s/%s/%s/%s/%s",
+		*targetPrefix,
+		strconv.Itoa(projectId),
+		inputManifest.Bucket,
+		inputManifest.PartitionDate,
+		"manifest.json")
+	manifestBytes, err := json.Marshal(outputManifest)
+	if err != nil {
+		return fmt.Errorf("Error Marshalling manifest: %w", err)
+	}
+
+	err = uploadToS3(cfg, ctx, bytes.NewReader(manifestBytes), *bucketName, manifestKey)
+	if err != nil {
+		return fmt.Errorf("Error uploading file to S3: %w", err)
+	}
+	log.Printf("Finished processing manifest and uploaded s3://%s/%s\n", *bucketName, manifestKey)
+
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.TODO()
+	cfg, err := config.LoadDefaultConfig(ctx)
+	cfg.Region = "eu-central-1"
+	if err != nil {
+		log.Println("Error:", err)
```

Comment:
Since this is server-side code, I'd recommend we use zerolog for all logging. It produces json structured output, which is way easier to handle. Also, it has a fatal level, which we can use instead of explicit return or os.exit.

Reply:
Done